droid-hal-device.inc

```diff
@@ -364,7 +368,33 @@ fi
 
 echo Building local tools
 ANDROID_ROOT=$(readlink -e %{android_root})
-(cd %{dhd_path}/helpers; make ANDROID_ROOT=$ANDROID_ROOT)
+
+if [ $ANDROID_VERSION_MAJOR -ge "8" ]; then
+    $ANDROID_ROOT/build/make/tools/fs_config/fs_config_generator.py aidarray \
+        $ANDROID_ROOT/system/core/libcutils/include/private/android_filesystem_config.h \
+        > generated_android_filesystem_config.h
+else
+    echo "#include <private/android_filesystem_config.h>;" > generated_android_filesystem_config.h
+fi
+
+# files have changed in android >=8 so we need to supply different ones to the
+# makefile. we also need the cstring workaround thus handling both issues here
+# at once
+if [ $ANDROID_VERSION_MAJOR -ge "8" ]; then
+    # sparse_read.cpp doesn't include cstring and therefore fails for us
+    # workaround:
+    echo "#include <cstring>" > $ANDROID_ROOT/system/core/libsparse/sparse_read_fix.cpp
+    echo "#include \"sparse_read.cpp\"" >> $ANDROID_ROOT/system/core/libsparse/sparse_read_fix.cpp
+    IMG2SIMG_SOURCES="backed_block.c output_file.c sparse.c sparse_crc32.c sparse_err.c sparse_read_fix.cpp img2simg.c ../base/stringprintf.cpp"
+    SIMG2IMG_SOURCES="backed_block.c output_file.c sparse.c sparse_crc32.c sparse_err.c sparse_read_fix.cpp simg2img.c ../base/stringprintf.cpp"
+else
+    IMG2SIMG_SOURCES="backed_block.c output_file.c sparse.c sparse_crc32.c sparse_err.c sparse_read.c img2simg.c"
+    SIMG2IMG_SOURCES="backed_block.c output_file.c sparse.c sparse_crc32.c sparse_err.c sparse_read.c simg2img.c"
+fi
+
+pushd %{dhd_path}/helpers
+make ANDROID_ROOT=$ANDROID_ROOT IMG2SIMG_SOURCES="$IMG2SIMG_SOURCES" SIMG2IMG_SOURCES="$SIMG2IMG_SOURCES"
```

Comment:
Minor: some adaptations override the .mk files, but mkbootimg is the only I could quick-find https://github.com/mer-hybris/droid-hal-mako/blob/master/helpers/mkbootimg.mk (yet still good to bear in mind we might have a porter or two knocking at door due to this API change :) )

Reply:
@sledges as far as i can tell overwriting *.mk files should still work :). If issues arise we need to tackle those later yes :).